airflow/utils/log/file_task_handler.py

```diff
@@ -261,3 +202,85 @@ def _init_file(self, ti):
                 logging.warning("OSError while change ownership of the log file")
 
         return full_path
+
+
+def read_local_log(location: str):
```

Comment:
This module has a mixed responsibility - It handles local files, remote files and k8s logs. I think we should divide it and move some part of the code to [`log_reader.py`](https://github.com/apache/airflow/blob/master/airflow/utils/log/log_reader.py), so that these functions can also be used by other handlers.